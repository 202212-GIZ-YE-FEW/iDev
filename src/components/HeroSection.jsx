import Image from "next/image";
import Link from "next/link";
import { withTranslation } from "next-i18next";

import Button from "./ui/Button";

function HeroSection({ t }) {
    const style = {
        backgroundImage: "url(/header.png)",
        minHeight: "calc(100vh - 5rem)",
    };
    return (
        <header style={style} className='min-w-[96vw] bg-cover'>
            <div className='container h-full flex flex-col justify-center lg:flex-row gap-y-10 lg:justify-between items-center'>
                <div className='flex flex-col justify-center text-center lg:text-start'>
                    <h1 className='text-xl sm:text-3xl lg:text-5xl uppercase'>
                        {t("weAreHereTo")}
                    </h1>
                    <p className='text-3xl sm:text-5xl lg:text-9xl mb-3 lg:mb-0 uppercase'>
                        {t("help")}
                    </p>
                    <Link
                        className='w-fit xl:w-full rtl:w-full'
                        href='/appointment'
                    >
                        <Button
                            content={t("bookAppointment")}
                            text-transform='uppercase'
                            filled='true'
                            size='large'
                            fontSize='text-lg md:text-xl lg:text-2xl'
                            radius='md'
                        />
                    </Link>
                </div>
                <div className='basis-1/2'>
                    <Image
                        src='/hero.png'
                        width={500}
                        height={200}
                        alt=''
                        className='max-w-full lg:min-w-[20rem] lg:max-w-[27rem] xl:min-w-[40rem]'
                    />
                </div>
            </div>
        </header>
    );
}
export default withTranslation("home")(HeroSection);
