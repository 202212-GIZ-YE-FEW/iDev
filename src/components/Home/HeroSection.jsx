import Image from "next/image";

import Button from "../ui/Button";

export default function Hero(prop) {
    const { t } = prop;
    const style = {
        backgroundImage: "url(/header.png)",
        backgroundSize: "cover",
        minWidth: "96.5vw",
        position: "relative",
        left: "0",
        right: "0",
        minHeight: "calc(100vh - 5rem)",
    };
    return (
        <header
            style={style}
            className='flex flex-col items-center justify-center'
        >
            <div className='container flex flex-col items-center justify-center md:flex-row md:space-s-20 space-y-10'>
                <div className='flex flex-col'>
                    <p className='text-xl md:text-2xl lg:text-3xl uppercase'>
                        we are here to
                    </p>
                    <span className='text-3xl md:text-5xl lg:text-9xl mb-3 lg:mb-0 uppercase'>
                        help
                    </span>
                    <Button
                        content='book an appointment'
                        text-transform='uppercase'
                        filled='true'
                        size='large'
                        fontSize='text-lg md:text-xl lg:text-2xl'
                        radius='md'
                    />
                </div>
                <div>
                    <Image
                        src='/hero.png'
                        className='w-full'
                        width={600}
                        height={500}
                        alt='hero image'
                    />
                </div>
            </div>
        </header>
    );
}
