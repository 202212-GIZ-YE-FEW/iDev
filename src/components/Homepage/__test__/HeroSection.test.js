import renderer from "react-test-renderer";

import HeroSection from "../HeroSection";

it("renders correctly", () => {
    const tree = renderer.create(<HeroSection />).toJSON();
    expect(tree).toMatchSnapshot();
});
