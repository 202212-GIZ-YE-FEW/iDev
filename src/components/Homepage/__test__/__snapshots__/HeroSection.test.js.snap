// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="min-w-[96vw] bg-cover"
  style={
    {
      "backgroundImage": "url(/header.png)",
      "minHeight": "calc(100vh - 5rem)",
    }
  }
>
  <div
    className="container h-full flex flex-col lg:flex-row gap-y-10 justify-between lg:mt-32"
  >
    <div
      className="flex flex-col mt-10 lg:mt-0 text-center lg:text-start"
    >
      <h1
        className="text-xl sm:text-3xl lg:text-5xl uppercase"
      >
        weAreHereTo
      </h1>
      <p
        className="text-3xl sm:text-5xl lg:text-9xl mb-3 lg:mb-0 uppercase"
      >
        help
      </p>
      <a
        className="w-fit xl:w-full rtl:w-full self-center lg:self-start"
        href="/appointment"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="relative self-center"
        >
          <button
            className="transform hover:bg-yellow transition duration-500  hover:animate-bounce hover:scale-75 active:bg-cyan focus:outline-none focus:ring focus:ring-cyan
                font-normal whitespace-nowrap cursor-pointer  capitalize text-lg md:text-xl lg:text-2xl rounded-md px-[25px] md:px-[35px] py-[8px] font-medium bg-cyan text-black  bg-light-gray text-gray"
            disabled={true}
            onClick={[Function]}
          >
            bookAppointment
          </button>
        </div>
      </a>
    </div>
    <div
      className="basis-1/2"
    >
      <img
        alt=""
        className="max-w-full lg:min-w-[20rem] lg:max-w-[27rem] xl:min-w-[40rem]"
        data-aos="zoom-in-up"
        data-nimg="1"
        decoding="async"
        height={200}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fhero.png&w=1080&q=75"
        srcSet="/_next/image?url=%2Fhero.png&w=640&q=75 1x, /_next/image?url=%2Fhero.png&w=1080&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={500}
      />
    </div>
  </div>
</header>
`;
